lib/net/modules/dns.toit

```diff
@@ -57,7 +57,60 @@ DEFAULT_CLIENT ::= DnsClient [
 // are separated by slash (/) in the key.
 ALL_CLIENTS_ ::= {DEFAULT_CLIENT.servers_.join "/": DEFAULT_CLIENT}
 
-default_client := DEFAULT_CLIENT
+user_set_client_/DnsClient? := null
+dhcp_client_/DnsClient? := null
+
+/**
+On Unix systems the default client is one that keeps an eye on changes in
+  /etc/resolv.conf.
+On FreeRTOS systems the default client is set by DHCP.
+On Windows we currently default to using Google and Cloudflare DNS servers.
+On all platforms you can set a custom default client with the
+  $(default_client= client) setter.
+*/
+default_client -> DnsClient:
+  if user_set_client_: return user_set_client_
+  if platform == "FreeRTOS" and dhcp_client_: return dhcp_client_
+  if platform == "Linux" or platform == "macOS": return etc_resolv_client_
+  return DEFAULT_CLIENT
+
+default_client= client/DnsClient -> none:
+  user_set_client_ = client
+
+current_etc_resolv_client_/DnsClient? := null
+etc_resolv_update_time_/Time? := null
+
+etc_resolv_client_ -> DnsClient:
+  error := catch --trace:
```

Comment:
What do you use the error for? I guess you want to make sure you always get a DnsClient back.